### ISSUE TITLE
@relationship fields cause similarly named fields with scalar types to be dropped during creation

### ISSUE THREAD

**Issue by keanugrieves** [neo4j/graphql#268]
**Describe the bug**
In custom resolvers, data in fields that start with the same name as a `@relationship` field are dropped before getting written to the database.

**Type definitions**
For an abbreviated example, with the following type defs...
```
type Mutation {
  registerUser(input: RegisterUserInput!): JWT!
}

input RegisterUserInput {`
  email: String!
  password: String!
  nameFirst: String!
  nameLast: String!
}

type User {
  id: ID @id(autogenerate: false)
  createdAt: DateTime
  updatedAt: DateTime
  updatedByRobotsAt: DateTime
  todosAssignedByRobotsAt: String
  email: String!
  lastLoginAt: DateTime!
  password: String! @private
  passwordResetCode: String @private
  passwordResetExpiresAt: DateTime @private
  passwordResetRequired: Boolean @private
  passwordResetRequiredReason: UserAuthPasswordResetRequiredReason @private
  role: UserAuthRole!
  firestoreExportedAt: DateTime @private
  firestoreId: ID @private
  nameFirst: String!
  nameLast: String!
  todosAssignedBy: [Todo!]!
    @relationship(type: "TODO__USER__ASSIGNED_BY", direction: IN)
  todosAssignedTo: [Todo!]!
    @relationship(type: "TODO__USER__ASSIGNED_TO", direction: IN)
  createdBy: User @relationship(type: "USER__USER__CREATED_BY", direction: OUT)
  updatedBy: User @relationship(type: "USER__USER__UPDATED_BY", direction: OUT)
}
```
...and a custom resolver that sets `id`, `createdAt`, `updatedAt`, `todosAssignedByRobotsAt`, and `updatedByRobotsAt` regardless of the input, `todosAssignedByAdminReason` and `updatedByRobotsAt` are dropped as properties during the custom `RegisterUser` mutation.

**To Reproduce**
See above.

**Expected behavior**
The data should be stored as properties on the node.

**Logs**
```
  @neo4j/graphql:graphql Incoming GraphQL:
Query:
mutation ($input: [UserCreateInput!]!) {
  createUsers(input: $input) {
    users {
      id
      todosAssignedByRobotsAt
      email
      password
      passwordResetCode
      passwordResetRequired
      firestoreId
      nameFirst
      nameLast
      passwordResetRequiredReason
      role
      createdAt
      updatedAt
      updatedByRobotsAt
      lastLoginAt
      passwordResetExpiresAt
      firestoreExportedAt
    }
  }
}
Variables:
{
  "input": [
    {
      "id": "472053006871623881366028",
      "todosAssignedByRobotsAt": "TODAY",
      "email": "sean@test.com",
      "password": "$2b$10$9Ssy6Jrakv34TpGQfxQqlOGDbETSLf5jrgA17ziiiMAcCAuF/aZAu",
      "nameFirst": "Sean",
      "nameLast": "Hanson",
      "role": "USER",
      "createdAt": {
        "year": 2021,
        "month": 6,
        "day": 16,
        "hour": 15,
        "minute": 9,
        "second": 26,
        "nanosecond": 29000000,
        "timeZoneOffsetSeconds": -25200
      },
      "updatedAt": {
        "year": 2021,
        "month": 6,
        "day": 16,
        "hour": 15,
        "minute": 9,
        "second": 26,
        "nanosecond": 29000000,
        "timeZoneOffsetSeconds": -25200
      },
      "updatedByRobotsAt": {
        "year": 2000,
        "month": 1,
        "day": 1,
        "hour": 0,
        "minute": 0,
        "second": 0,
        "nanosecond": 0,
        "timeZoneOffsetSeconds": -28800
      },
      "lastLoginAt": {
        "year": 2021,
        "month": 6,
        "day": 16,
        "hour": 15,
        "minute": 9,
        "second": 26,
        "nanosecond": 31000000,
        "timeZoneOffsetSeconds": -25200
      }
    }
  ]
} +128ms
  @neo4j/graphql:auth JWT not configured +129ms
  @neo4j/graphql:execute About to execute Cypher:
Cypher:
CALL {
CREATE (this0:User)
SET this0.id = $this0_id
SET this0.email = $this0_email
SET this0.password = $this0_password
SET this0.nameFirst = $this0_nameFirst
SET this0.nameLast = $this0_nameLast
SET this0.role = $this0_role
SET this0.createdAt = $this0_createdAt
SET this0.updatedAt = $this0_updatedAt
SET this0.lastLoginAt = $this0_lastLoginAt
RETURN this0
}

RETURN
this0 { .id, .todosAssignedByRobotsAt, .email, .password, .passwordResetCode, .passwordResetRequired, .firestoreId, .nameFirst, .nameLast, .passwordResetRequiredReason, .role, createdAt: apoc.date.convertFormat(toString(this0.createdAt), "iso_zoned_date_time", "iso_offset_date_time"), updatedAt: apoc.date.convertFormat(toString(this0.updatedAt), "iso_zoned_date_time", "iso_offset_date_time"), updatedByRobotsAt: apoc.date.convertFormat(toString(this0.updatedByRobotsAt), "iso_zoned_date_time", "iso_offset_date_time"), lastLoginAt: apoc.date.convertFormat(toString(this0.lastLoginAt), "iso_zoned_date_time", "iso_offset_date_time"), passwordResetExpiresAt: apoc.date.convertFormat(toString(this0.passwordResetExpiresAt), "iso_zoned_date_time", "iso_offset_date_time"), firestoreExportedAt: apoc.date.convertFormat(toString(this0.firestoreExportedAt), "iso_zoned_date_time", "iso_offset_date_time") } AS this0
Params:
{
  "this0_id": "472053006871623881366028",
  "this0_email": "sean@test.com",
  "this0_password": "$2b$10$9Ssy6Jrakv34TpGQfxQqlOGDbETSLf5jrgA17ziiiMAcCAuF/aZAu",
  "this0_nameFirst": "Sean",
  "this0_nameLast": "Hanson",
  "this0_role": "USER",
  "this0_createdAt": {
    "year": 2021,
    "month": 6,
    "day": 16,
    "hour": 15,
    "minute": 9,
    "second": 26,
    "nanosecond": 29000000,
    "timeZoneOffsetSeconds": -25200
  },
  "this0_updatedAt": {
    "year": 2021,
    "month": 6,
    "day": 16,
    "hour": 15,
    "minute": 9,
    "second": 26,
    "nanosecond": 29000000,
    "timeZoneOffsetSeconds": -25200
  },
  "this0_lastLoginAt": {
    "year": 2021,
    "month": 6,
    "day": 16,
    "hour": 15,
    "minute": 9,
    "second": 26,
    "nanosecond": 31000000,
    "timeZoneOffsetSeconds": -25200
  }
} +67ms
  @neo4j/graphql:execute Execute successful, received 1 records +55ms
```

**System (please complete the following information):**

-   OS: Windows 10
-   Version: @neo4j/graphql^1.0.2, @neo4j/graphql-ogm^1.0.2
-   Node.js version: 12.22.1

**Additional context**
I haven't tried reproducing these steps by actually connecting the `@relationship` fields to other records, but on optional matches, this would still be an issue.

**Comment by darrellwarde** [comment]
This is going to be fixed in version 2.0.0, and the fix for version 1.x isn't so great, so we'll hold off until the 2.0.0 release for this one.

**Comment by darrellwarde** [comment]
In an effort to clean our issue backlog a bit, going to close this one off as the fix is available in any 2.0.0 release whenever you decide to upgrade, be it during prerelease or after stable.
